pkg/kubelet/metrics/metrics.go

```diff
@@ -219,6 +238,40 @@ var (
 		},
 		[]string{"operation_type"},
 	)
+
+	// RuntimeImagePullDuration is a Histogram that tracks the duration (in seconds) for image pull
+	RuntimeImagePullDuration = metrics.NewHistogram(
+		&metrics.HistogramOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           ImagePullDurationKey,
+			Help:           "Cumulative number of image pull duration by image spec",
+			Buckets:        podLatencyBuckets,
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// RuntimeImagePullCount is a Counter that tracks kubelet's attempt to ensure image.
+	// Two cases are excluded: 1) image not present AND image pull policy is NEVER;
+	// 2) image shall be pulled but should backoff
+	RuntimeImagePullCount = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           ImageEnsureCountKey,
+			Help:           "Count of the number of time kubelet tries to ensure image, except: 1) image not present AND image pull policy is NEVER; 2) image shall be pulled but should backoff",
```

Comment:
Should we combine this and the metric below, and make `cache_hit` a label?

Reply:
my concern here is that, since this metric contains "image shall be pulled but should backoff", if we combine this with cache_hit, glitches at image registry might cause decrease in cache_hit which might be confusing.

Reply:
Ah. I see.  Would it be better not to include that case ("image shall be pulled but should backoff"), and just count actual pull attempts?

Otherwise, I am fine with the current metrics.

Reply:
yeah that's possible, lemme start with excluding the "image shall be pulled but should backoff" case. will let you know when I have the updates ready.